nullaway/src/test/resources/com/uber/nullaway/testdata/NullAwayTryFinallyCases.java

```diff
@@ -358,4 +349,90 @@ public void tryMultipleCatch(@Nullable Object o) {
       }
     }
   }
+
+  private boolean nestedCFGConstructionTest(Object o) {
+    boolean result = true;
+    java.io.BufferedWriter out = null;
+    try {
+      try {
+      } finally {
+        out = new java.io.BufferedWriter(new java.io.OutputStreamWriter(System.err));
+      }
+      if (o != null) {
+        out.write(' ');
+      }
+    } catch (Exception e) {
+    } finally {
+    }
+    return result;
+  }
+
+  private boolean nestedCFGConstructionTest2() throws IOException {
+    java.io.BufferedWriter out =
+        new java.io.BufferedWriter(new java.io.OutputStreamWriter(System.err));
+    try {
+      try {
+        return true;
+      } finally {
+      }
+    } finally {
+      out.write(' ');
+      out.close();
+    }
+  }
+
+  class IndirectInitialization {
+    Object f;
+
+    IndirectInitialization(Object o1) {
+      // Do or do not...
+      init();
+    }
+
+    IndirectInitialization(Object o1, Object o2) {
+      // ... but here there is try
+      try {
+        init();
+      } finally {
+      }
+    }
+
+    private void init() {
+      this.f = new Object();
+    }
+  }
+
+  class IndirectInitialization2 {
+    Object f;
+
+    //// Fixme: This should work recursivelly
```

Comment:
This is actually not-recursive right now, regardless of `try/finally`. Can fix it, but maybe worth doing a different PR for that :)